airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValue.kt

```diff
@@ -48,6 +51,24 @@ sealed interface AirbyteValue {
                         "Unrecognized value (${value.javaClass.name}: $value"
                     )
             }
+
+        fun airbyteTypeOf(value: AirbyteValue): AirbyteType =
+            when (value) {
+                // ArrayType requires the type of the object which we do not convey
+                is ArrayValue -> TODO()
+                is BooleanValue -> BooleanType
+                is DateValue -> DateType
+                is IntegerValue -> IntegerType
+                // Null is awkward because the value doesn't track the actual type information
+                is NullValue -> TODO()
+                is NumberValue -> NumberType
+                is ObjectValue -> ObjectTypeWithoutSchema
```

Comment:
Why are we using the WithoutType here? It could also be a UnionType, I'm not sure where those will fit.